app/controlplane/pkg/data/workflow.go

```diff
@@ -317,8 +293,8 @@ func entWFToBizWF(ctx context.Context, w *ent.Workflow, r *ent.WorkflowRun) (*bi
 		wf.ContractRevisionLatest = lv.Revision
 	}
 
-	if r != nil {
-		lastRun, err := entWrToBizWr(ctx, r)
+	if latestRun := w.Edges.LatestWorkflowRun; latestRun != nil {
```

Comment:
should we query if it's not here? We do that pattern in some other places, either take it from the preloaded edge or load it?

Reply:
When I removed the workflowRun option from `entWFToBizWF`, I added `WithLatestWorkflowRun()` to all the places where we were previously fetching the latest workflow run on demand. This ensures it works the same as before. My concern with always fetching the latest workflow run when not found on the edge is that it could lead to unnecessary queries since the logic doesn't always use it.

I prefer to keep it this way, and we can update the query if we need the latest workflow run in the future.